src/main/java/jp/ikedam/jenkins/plugins/groovy_label_assignment/GroovyLabelAssignmentAction.java

```diff
@@ -32,16 +33,39 @@
  */
 public class GroovyLabelAssignmentAction implements LabelAssignmentAction
 {
-    private Label label;
+    @Deprecated
+    transient private Label label;
+    
+    private final String labelString;
     
     /**
      * Constructor
      * 
-     * @param label assigned label.
+     * @param labelString assigned label expression.
+     * @since 1.1.1
      */
+    public GroovyLabelAssignmentAction(String labelString)
+    {
+        this.labelString = labelString;
+    }
+    
+    /**
+     * @param label
+     * @deprecated use {@link GroovyLabelAssignmentAction#GroovyLabelAssignmentAction(String)}
+     */
+    @Deprecated
     public GroovyLabelAssignmentAction(Label label)
     {
-        this.label = label;
+        this(label.getExpression());
+    }
+    
+    private Object readResolve()
+    {
+        if(label != null)
+        {
+            return new GroovyLabelAssignmentAction(label.getExpression());

```

Comment:
I soppose you don't know `readResolve`
https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility

I provide this method to allow users upgrade `GroovyLabelAssignmentAction` created with older versions of groovy-label-assignment-plugin.
Consequently, `readResolve` HAVE TO access deprecated fields to retrieve values created in the older version.
That's logical.

`readResolve` is called by [xstream](https://wiki.jenkins-ci.org/display/JENKINS/Architecture).
I believe xstream call it even `private`: http://x-stream.github.io/faq.html#Serialization